src/test/java/com/google/devtools/build/lib/bazel/repository/skylark/SkylarkRepositoryContextTest.java

```diff
@@ -278,6 +281,44 @@ public void testRead() throws Exception {
     }
   }
 
+  @Test
+  public void testPatch() throws Exception {
+    setUpContexForRule("test");
+    SkylarkPath foo = context.path("foo");
+    context.createFile(foo, "line one\n", false, true, null);
+    SkylarkPath patchFile = context.path("my.patch");
+    context.createFile(context.path("my.patch"),
+        "--- foo\n" +
+        "+++ foo\n" +
+        "@@ -1,1 +1,2 @@\n" +
+        " line one\n" +
+        "+line two\n", false, true, null);
+    context.patch(patchFile, 0, null);
+    testOutputFile(foo.getPath(), String.format("line one%nline two%n"));
+  }
+
+  @Test
+  public void testPatchFailed() throws Exception {
```

Comment:
Great! Please add two more tests:  1,4 (3 is probably similar to 2)

https://github.com/bazelbuild/bazel/pull/8555/files/dc0dccc8bbc3e676695fc114aa20f0e921bca2ba#r299372582

Reply:
Done